src/som/interpreter/nodes/IsValueCheckNode.java

```diff
@@ -82,8 +81,12 @@ private void dropCheckNode() {
   private static final class ValueCheckNode extends IsValueCheckNode {
     ValueCheckNode(final ExpressionNode self) {
       super(self);
+      thrower = new ExceptionSignalingNode("NotAValue",
+          "signalWith:", self.sourceSection);
     }
 
+    @Child protected ExceptionSignalingNode thrower;
```

Comment:
Fields are ideally defined before the constructors

Reply:
Done.